receiver/prometheusreceiver/metrics_receiver.go

```diff
@@ -347,9 +369,135 @@ func (r *pReceiver) initPrometheusComponents(ctx context.Context, logger log.Log
 			r.settings.TelemetrySettings.ReportStatus(component.NewFatalErrorEvent(err))
 		}
 	}()
+
 	return nil
 }
 
+func (r *pReceiver) initAPIServer(ctx context.Context, host component.Host) error {
+	r.settings.Logger.Info("Starting Prometheus API server")
+
+	o := &web.Options{
+		ScrapeManager: r.scrapeManager,
+		Context:       ctx,
+		ListenAddress: r.cfg.APIServer.ServerConfig.Endpoint,
+		ExternalURL: &url.URL{
+			Scheme: "http",
+			Host:   r.cfg.APIServer.ServerConfig.Endpoint,
+			Path:   "",
+		},
+		RoutePrefix:    "/",
+		ReadTimeout:    time.Minute * readTimeoutMinutes,
+		PageTitle:      "Prometheus Receiver",
+		Flags:          make(map[string]string),
+		MaxConnections: maxConnections,
+		IsAgent:        true,
+		Registerer:     r.registerer,
+		Gatherer:      prometheus.DefaultGatherer,
```

Comment:
You can change r.registerer to a *prometheus.Registry, which implements the Gatherer and Registerer interfaces, and then use it here.  Maybe rename to r.registry as well